lib/tapioca/generator.rb

```diff
@@ -108,6 +108,52 @@ def build_todos
       say("Please review changes and commit them.", [:green, :bold])
     end
 
+    sig { params(requested_constants: String).void }
+    def build_dsl(*requested_constants)
+      base_directory = Pathname.new("#{Config::DEFAULT_RBIDIR}/dsl")
+      generation_command_template = config.generate_command
+
+      loader = Loader.new(Gemfile.new)
+      loader.load_rails(
+        environment_load: true,
+        eager_load: requested_constants.empty?
+      )
+
+      Dir["#{Config::TAPIOCA_PATH}/generators/**/*.rb"].each do |generator|
+        require File.expand_path(generator)
+      end
+
+      Dir["#{__dir__}/compilers/dsl/*.rb"].each do |generator|
+        require File.expand_path(generator)
+      end
+
+      compiler = Compilers::DslCompiler.new(
+        requested_constants: requested_constants.map(&:constantize),
+        requested_generators: ENV.fetch("GENERATORS", "").split(",").map(&:strip)
+      )
+
+      compiler.run do |constant, contents|
+        next if contents.nil?
+
+        command = format(generation_command_template, constant.name)
+        constant_name = Module.instance_method(:name).bind(constant).call
+        filename = base_directory.join(constant_name.underscore + ".rbi")
+
+        contents = <<~RBI
+          # DO NOT EDIT MANUALLY
```

Comment:
We're decoupling this in each new method generating a different kind of RBI. I think we can move it to the compiler itself.